apps/meteor/client/sidebar/RoomMenu.tsx

```diff
@@ -137,16 +141,35 @@ const RoomMenu = ({ rid, unread, threadUnread, alert, roomOpen, type, cl, name =
 		);
 	});
 
+	const handleThreadsToBeRead = useMutableCallback(async () => {
+		try {
+			if (threadUnread) {
+				const { threads } = await getThreadsList({
+					rid,
+					type: 'unread',
+					count: 0,
+					offset: 0,
+				});
+				const promises = threads.map((thread) => readThreads(thread?._id));
```

Comment:
I'm wondering here if this should be front-end responsibility... Perhaps we should update the backend `readMessages` function to do this? Or even create a new endpoint for it if that's the case.